docs/upgrading/v6-data.md

```diff
@@ -0,0 +1,216 @@
+  return (
+    <Routes>
+      <Route path="/" element={<Home />} />
+      <Route path="/blog/*" element={<BlogApp />} />
+      <Route path="/users/*" element={<UserApp />} />
+    </Routes>
+  );
+}
+
+function Home() {
+  /* Unchanged */
+}
+function BlogApp() {
+  /* Unchanged */
+}
+function UserApp() {
+  /* Unchanged */
+}
+```
+
+🥳 Congrats - you're now rendering a data router app! But wait a minute - we can't use any of the new stuff yet since none of our routes are defined at the top with `createBrowserRouter` 😢. To access the new APIs, we need to start lifting routes one-by-one to the data router.
+
+### Start lifting routes and leveraging the data APIs
+
+Let's start with the `/` route for the `<Home>` component. All we need to do is lift the `<Route>` definition up to the data router:
+
+```tsx lines=[2,13]
+const router = createBrowserRouter([
+  { path: "/", Component: Home }, // 🆕
+  { path: "*", Component: Root },
+]);
+
+export default function App() {
+  return <RouterProvider router={router} />;
+}
+
+function Root() {
+  return (
+    <Routes>
+      {/* ⬆️ Home route lifted up to the data router */}
+      <Route path="/blog/*" element={<BlogApp />} />
+      <Route path="/users/*" element={<UserApp />} />
+    </Routes>
+  );
+}
+```
+
+Now you can add data APIs to your home route (`loader`, `action`, `errorElement`) and start leveraging data hooks inside your Home component (`useLoaderData`, `useActionData`, `useFetcher`, `<Form>`, etc.).
+
+Now let's look at lifting the Blog App upwards, but still doing it one leaf route at a time. In order to lift the `/blog` index route up, we need the `/blog/*` splat route lifted as well, but we can still render the `/blog/posts` route where it is and do that separately.
+
+```tsx lines=[3-12,23,32]
+const router = createBrowserRouter([
+  { path: "/", Component: Home },
+  {
+    // Lifted blog splat route
+    path: "/blog/*",
+    children: [
+      // New blog index route
+      { index: true, Component: () => <h1>Blog Index</h1> },
+      // Blog subapp splat route added for /blog/posts matching
+      { path: "*", Component: BlogApp },
+    ],
+  },
+  { path: "*", Component: Root },
+]);
+
+export default function App() {
+  return <RouterProvider router={router} />;
+}
+
+function Root() {
+  return (
+    <Routes>
+      {/* ⬆️ Blog splat route lifted */}
+      <Route path="/users/*" element={<UserApp />} />
+    </Routes>
+  );
+}
+
+function BlogApp() {
+  return (
+    <Routes>
+      {/* ⬆️ Blog index route lifted */}
+      <Route path="posts" element={<h1>Blog Posts</h1>} />
+    </Routes>
+  );
+}
+```
+
+And now your blog index route can participate in data loading.
+
+You can keep doing this one route at a time until you've eventually converted all of your routes to data routes and can no longer use any nested `<Routes>` to define your routing tree. To avoid bundle bloat, it's recommended to leverage the [route.lazy][route-lazy] prop to lazily load your routes.
```

Comment:
offtopic: i wish vite didn't have problems with bundling css to make lazy loading routes simpler 😓 

https://github.com/vitejs/vite/issues/3924